### ISSUE TITLE
Add config option to log in json format

### ISSUE THREAD

**Issue by mieliespoor** [DependencyTrack/dependency-track#2517]
### Current Behavior

The logs are current written in text format.

```
2023-02-23 03:46:42,749 INFO [ProjectMetricsUpdateTask] Executing metrics update for project d0cd8a11-b43b-4c58-a08a-a80f5162d97d
```

### Proposed Behavior

Json log entries helps making it easier to search the logs - especially when the logs are ingested with AWS Cloudwatch. Taking the above log entry, it could be transformed to the following:

```
{
    "level": "INFO",
    "task": ProjectMetricsUpdateTask,
    "msg": "Executing metrics update for project",
    "time": "2023-02-23T03:46:42.749Z",
    "project": "d0cd8a11-b43b-4c58-a08a-a80f5162d97d"
}
```

The additional problem with text logging into something like cloudwatch is that stack traces being logged by this tool are written as individual log entries which could be counter productive. This could instead be written as follows:

```
{
    "level": "INFO",
    "type": "LoggableUncaughtExceptionHandler",
    "msg": "An unknown error occurred in an asynchronous event or notification thread",
    "time": "2023-02-23T04:12:38.164Z",
    "project": "d0cd8a11-b43b-4c58-a08a-a80f5162d97d",
    "exceptionMessage": "org.apache.lucene.store.AlreadyClosedException: Underlying file changed by an external force at 2023-02-23T00:45:43.98Z, (lock=NativeFSLock(path=/data/.dependency-track/index/vulnerability/write.lock,impl=sun.nio.ch.FileLockImpl[0:9223372036854775807 exclusive valid],creationTime=2023-02-22T21:45:44.413Z))",
    "stacktrace": "	at org.apache.lucene.store.NativeFSLockFactory$NativeFSLock.ensureValid(NativeFSLockFactory.java:191)\nat org.apache.lucene.store.LockValidatingDirectoryWrapper.createOutput(LockValidatingDirectoryWrapper.java:43)at org.apache.lucene.store.TrackingDirectoryWrapper.createOutput(TrackingDirectoryWrapper.java:43)\nat org.apache.lucene.codecs.compressing.CompressingStoredFieldsWriter.<init>(CompressingStoredFieldsWriter.java:121)\nat org.apache.lucene.codecs.compressing.CompressingStoredFieldsFormat.fieldsWriter(CompressingStoredFieldsFormat.java:130)\nat org.apache.lucene.codecs.lucene87.Lucene87StoredFieldsFormat.fieldsWriter(Lucene87StoredFieldsFormat.java:141)\nat org.apache.lucene.index.StoredFieldsConsumer.initStoredFieldsWriter(StoredFieldsConsumer.java:48)\nat org.apache.lucene.index.StoredFieldsConsumer.startDocument(StoredFieldsConsumer.java:55)\nat org.apache.lucene.index.DefaultIndexingChain.startStoredFields(DefaultIndexingChain.java:452)\nat org.apache.lucene.index.DefaultIndexingChain.processDocument(DefaultIndexingChain.java:488)\nat org.apache.lucene.index.DocumentsWriterPerThread.updateDocuments(DocumentsWriterPerThread.java:208)\nat org.apache.lucene.index.DocumentsWriter.updateDocuments(DocumentsWriter.java:415)\nat org.apache.lucene.index.IndexWriter.updateDocuments(IndexWriter.java:1471)\nat org.apache.lucene.index.IndexWriter.updateDocument(IndexWriter.java:1757)\nat org.apache.lucene.index.IndexWriter.addDocument(IndexWriter.java:1400)\nat org.dependencytrack.search.VulnerabilityIndexer.add(VulnerabilityIndexer.java:78)\nat org.dependencytrack.search.VulnerabilityIndexer.add(VulnerabilityIndexer.java:44)\nat org.dependencytrack.tasks.IndexTask.inform(IndexTask.java:61)\nat alpine.event.framework.BaseEventService.lambda$publish$0(BaseEventService.java:101)\nat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)\nat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)\nat java.base/java.lang.Thread.run(Unknown Source)"
}
```

Keep in mind that the above "json" logs has been constructed with the hand and not a formal logging tool, so implemented results might look different, but the above at least shows the idea behind it.

### Checklist

- [X] I have read and understand the [contributing guidelines](https://github.com/DependencyTrack/dependency-track/blob/master/CONTRIBUTING.md#filing-issues)
- [X] I have checked the [existing issues](https://github.com/DependencyTrack/dependency-track/issues) for whether this enhancement was already requested

**Comment by nscuro** [comment]
We use [logback](https://logback.qos.ch/) for logging, with the default configuration being located [here](https://github.com/DependencyTrack/dependency-track/blob/master/src/main/docker/logback.xml).

Custom configurations can be provided via `-Dlogback.configurationFile=/path/to/logback.xml`. When using the container image distribution of the DT API server, that option may be passed via the `EXTRA_JAVA_OPTIONS` environment variable.

However, it doesn't look like logback supports JSON natively. So we will have to add a library like [logstash-logback-encoder](https://github.com/logfellow/logstash-logback-encoder) to provide that functionality.

I think it should be doable to ship an alternative logback config that uses JSON, and expose an additional environment variable for the container image to switch between both configs.

**Comment by nscuro** [comment]
Support for logging in JSON format has been implemented in Alpine (https://github.com/stevespringett/Alpine/pull/503), and will be shipped with Dependency-Track 4.9.0. The remaining task here is to add proper documentation for how to set it up.
